apis/apiextensions/v1alpha1/ccrd/crd.go

```diff
@@ -16,7 +16,7 @@ limitations under the License.
 
 // Package ccrd generates CustomResourceDefinitions from Crossplane definitions.
 //
-// v1beta1.JSONSchemaProps is incompatible with controller-tools (as of 0.2.4)
+// v1.JSONSchemaProps is incompatible with controller-tools (as of 0.2.4)
 // because it is missing JSON tags and uses float64, which is a disallowed type.
 // We thus copy the entire struct as CRDSpecTemplate. See the below issue:
 // https://github.com/kubernetes-sigs/controller-tools/issues/291
```

Comment:
It seems like this issue is closed with https://github.com/kubernetes-sigs/controller-tools/pull/427 . Would you mind testing if the new controller-tools can work for us here?

Reply:
@muvaf unfortunately we can't update to `v0.4.0`, which is the version that the referenced issue is fixed in because of the bug described in https://github.com/crossplane/crossplane/pull/1850 😞

Reply:
Yeah, this was the rabbit hole I went down. I was hoping we might get some kind of validation of this validation schema, but had no luck.